src/storage/mod.rs

```diff
@@ -86,80 +98,165 @@ pub enum StorageCb {
         lock_ts: u64,
+        /// The commit timestamp.
         commit_ts: u64,
     },
+    /// Rollback mutations on a single key.
+    ///
+    /// This should be following a [`Prewrite`](Command::Prewrite) on the given key.
     Cleanup {
         ctx: Context,
         key: Key,
+        /// The transaction timestamp.
         start_ts: u64,
     },
+    /// Rollback from the transaction that was started at `start_ts`.
+    ///
+    /// This should be following a [`Prewrite`](Command::Prewrite) on the given key.
     Rollback {
         ctx: Context,
         keys: Vec<Key>,
+        /// The transaction timestamp.
         start_ts: u64,
     },
+    /// Rollback pessimistic locks identified by `start_ts` and `for_update_ts`.
+    ///
+    /// This can roll back an [`AcquirePessimisticLock`](Command::AcquirePessimisticLock) command.
     PessimisticRollback {
         ctx: Context,
+        /// The keys to be rolled back.
         keys: Vec<Key>,
+        /// The transaction timestamp.
         start_ts: u64,
         for_update_ts: u64,
     },
+    /// Scan locks from `start_key`, and find all locks whose timestamp is before `max_ts`.
     ScanLock {
         ctx: Context,
+        /// The maximum transaction timestamp to scan.
         max_ts: u64,
+        /// The key to start from. (`None` means start from the very beginning.)
         start_key: Option<Key>,
+        /// The result limit.
         limit: usize,
     },
+    /// Resolve locks according to `txn_status`.
+    ///
+    /// During the GC operation, this should be called to clean up stale locks whose timestamp is
+    /// before safe point.
+    ///
     ResolveLock {
         ctx: Context,
+        /// Maps lock_ts to commit_ts. If a transaction was rolled back, it is mapped to 0.
+        ///
+        /// For example, let `txn_status` be `{ 100: 101, 102: 0 }`, then it means that the transaction
+        /// whose start_ts is 100 was committed with commit_ts `101`, and the transaction whose
+        /// start_ts is 102 was rolled back. If there are these keys in the db:
+        ///
+        /// * "k1", lock_ts = 100
+        /// * "k2", lock_ts = 102
+        /// * "k3", lock_ts = 104
+        /// * "k4", no lock
+        ///
+        /// Here `"k1"`, `"k2"` and `"k3"` each has a not-yet-committed version, because they have
+        /// locks. After calling resolve_lock, `"k1"` will be committed with commit_ts = 101 and `"k2"`
+        /// will be rolled back.  `"k3"` will not be affected, because its lock_ts is not contained in
+        /// `txn_status`. `"k4"` will not be affected either, because it doesn't have a non-committed
+        /// version.
         txn_status: HashMap<u64, u64>,
         scan_key: Option<Key>,
         key_locks: Vec<(Key, Lock)>,
     },
+    /// Resolve locks on `resolve_keys` according to `start_ts` and `commit_ts`.
+    ///
+    /// During the GC operation, this should be called to clean up stale locks whose timestamp is
```

Comment:
@zhangjinpeng1987 If I didn't got into mistake, I think this is not used for GC? PTAL

Reply:
Yes, it is triggered by read request.

Reply:
Removed this comment.